Add details about supported service types, copy edits (#10)
R-ArcGIS/r-bridge-site: location-services/read-data.qmd

 - [FeatureServer](https://developers.arcgis.com/rest/services-reference/enterprise/feature-service.htm)
    - [FeatureLayer](https://developers.arcgis.com/rest/services-reference/enterprise/feature-layer.htm)
    - [Table](https://developers.arcgis.com/rest/services-reference/enterprise/feature-layer.htm)
 - [MapServer](https://developers.arcgis.com/rest/services-reference/enterprise/map-service.htm)
    - [GroupLayer](https://developers.arcgis.com/web-map-specification/objects/groupLayer/)
 - [ImageServer](https://developers.arcgis.com/rest/services-reference/enterprise/image-service.htm)

Metadata for all of the above service types can be accessed using `arc_open()`. Feature data can be read in using `arc_select()` for FeatureLayer, Table, and ImageServer.

This tutorial will teach you the basics of reading data from hosted Feature Layers into R as [`{sf}`](https://r-spatial.github.io/sf/) objects using`{arcgislayers}`. 

## Objective

The objective of this tutorial is to teach you how to:

- find a Feature Layer url from ArcGIS Online
- read in the data from the Feature Layer

> - filter the Feature Layer data by column
> - filter the Feature Layer data by row

Reviewer: I think these need to be rephrased. `filter` is a row-wise operation where we select rows based on some logical predicates and `select` is a column-wise operation where we fetch specific columns based on named 
```suggestion
- select the Feature Layer data by column
- filter the Feature Layer data by attributes
```

Author: Thank you for the fix! the phrasing I had was not sitting super well with me either.

> - use `dplyr` for filtering

Reviewer: ```suggestion
- use `dplyr` for selecting and filtering
```

```

Then pass this variable to `arc_open()` and save it to `flayer` (feature layer).

```{r}
flayer <- arc_open(furl)
flayer
```

`arc_open()` will create a `FeatureLayer` object. Under the hood, this is really just a list containing the feature layer's metadata. 

:::{.callout-note collapse="true" title="FeatureLayer details for the curious"}
The `FeatureLayer` object is obtained by adding `?f=json` to the feature layer url and processing the json. All of the metadata is stored in the `FeatureLayer` object. You can see this by running `unclass(flayer)`. Be warned! It gets messy. 
:::

With this `FeatureLayer` object, you can read data from the service into R! 

## Reading from a Feature Layer

Once you have a `FeatureLayer` object, you can read its data into memory using the `arc_select()` function. By default, if you use `arc_select()` on a `FeatureLayer` without any additional arguments, the entire service will be brought into memory.

:::{.callout-warning}
Avoid reading in more data than you need! Reading an entire feature service is fine for datasets with fewer than 5,000 features. But when there are more than 10,000 features, performance and memory may be throttled. 

> Exceptionally detailed geometries require more data to be transferred across the web and may be slower to process.

Reviewer: ```suggestion
Exceptionally detailed geometries require more data to be transferred across the web and may be slower to process and may require adjustment of the `page_size` argument of `arc_select()`.
```
